lib/hsa/mcwamp_hsa.cpp

```diff
@@ -617,17 +618,26 @@ class HSABarrier : public HSAOp {
     int depCount;
     hc::memory_scope _acquire_scope;
 
+    // capture the state of _nextSyncNeedsSysRelease and _nextKernelNeedsSysAcquire after
+    // the barrier is issued.  Cross-queue synchronziation commands which synchronize
+    // with the barrier (create_blocking_marker) then can transer the correct "needs" flags.
+    bool                                            _barrierNextSyncNeedsSysRelease;
+    bool                                            _barrierNextKernelNeedsSysAcquire;
+
 public:
     uint16_t  header;  // stores header of AQL packet.  Preserve so we can see flushes associated with this barrier.
 
     // array of all operations that this op depends on.
     // This array keeps a reference which prevents those ops from being deleted until this op is deleted.
-    std::shared_ptr<HSAOp> depAsyncOps [HSA_BARRIER_DEP_SIGNAL_CNT];
+    std::shared_ptr<HSABarrier> depAsyncOps [HSA_BARRIER_DEP_SIGNAL_CNT];
```

Comment:
@bensander this means all ops (kernel/copy/barrier) would only depend on barrier ops, correct?